### ISSUE TITLE
Controlling transition progress with animated node.

### ISSUE THREAD

**Issue by osdnk** [software-mansion/react-native-reanimated#269]
Now it's possible to make only discrete changes which might be connected with some transitions which state is not controlled during performing. 
I see it might be useful to manage the state of animation (progress) with animated nodes. 

/@kmagiera @adamczyk777

**Comment by wcandillon** [comment]
Yes I agree, it would be so useful to have an animation value that goes from 0 to 1 that would represent the transition. The use case I have is that `transform` at the moment is not supported by the transition API. Having an animation value for that would solve that problem.

**Comment by wcandillon** [comment]
This is the kind of animation that could be built with the transition API is it would support transform or if we could follow the transition via an animated value: https://twitter.com/wcandillon/status/1142159471285800962

**Comment by wcandillon** [comment]
@osdnk I wrote this hook that allows me to fulfill this use case: https://github.com/kmagiera/react-native-reanimated/issues/321#issuecomment-504726195

**Comment by ShaMan123** [comment]
+1

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.
 
We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
